File: Front/src/app/shared/model/course.ts
import { User } from "./User";

export class Course  {
    name: string = "";
    description: string = "";
    startDate: Date = new Date();
    startDateString: string = "";
    numOfModules: number = 0;
    duration: number = 0;
    candidates: User[] = [];
    mentor: User = new User();
    mentorId: string = "";

    public constructor(obj?: any) {
        if (obj) {
          this.name = obj.name;
          this.description = obj.description;
          this.startDateString = obj.startDateString;
          this.startDate = obj.startDate;
          this.numOfModules = obj.numOfModules;
          this.candidates = obj.candidates;
          this.mentor = obj.mentor;
          this.mentorId = obj.mentorId;
        }
      }
}

File: Back/CourseManagementApp/Model/Course.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace CourseManagementApp.Model
{
    public class Course : IEntity<Guid>
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }
        [Required]
        [MaxLength(100)]
        public string Name { get; set; }
        [Required]
        [MaxLength(1000)]
        public string Description { get; set; }
        [Required]
        public int NumOfModules { get; set; }
        [Required]
        public DateTime EndDate { get; set; }
        [Required]
        public int Duration { get; set; }
        public User Mentor { get; set; }
        public string MentorId { get; set; }
        public virtual ICollection<CandidateCourse> CandidatesEnrolled { get; set; }  
        public Training Training { get; set; }
    }
}
